util/struct.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type StructName string
+
+type StructFields []StructField
+
+// StructField contains information about a single field in a struct.
+type StructField struct {
+	Name  string
+	Value string
+	Tag   string
+}
+
+func (sf *StructFields) Append(name StructName, value, tag string) {
```

Comment:
> Suggest making tag implicitly generated based on `name`? so the caller just need to provide `name` and `value`.

Updated.

> Make sure the tag is `camelCase` not just the first letter lower case. There should be some lib able to reuse directly.

I wasn't able to find a suitable library. However, since the name is defined by us so I think it should be ok for now, WDYT?

Reply:
It's OK.